server/src/test/java/keywhiz/service/permission/AlwaysAllowDelegatingPermissionCheckTest.java

```diff
@@ -30,42 +29,71 @@ public class AlwaysAllowDelegatingPermissionCheckTest {
   AlwaysAllowDelegatingPermissionCheck alwaysAllow;
   KeywhizPrincipal principal;
   Objects target;
+  String isAllowedSuccessMetricName;
+  String isAllowedFailureMetricName;
+  String checkAllowedOrThrowSuccessMetricName;
+  String checkAllowedOrThrowExceptionMetricName;
 
   @Before
   public void setUp() {
     metricRegistry = new MetricRegistry();
     alwaysAllow = new AlwaysAllowDelegatingPermissionCheck(metricRegistry, delegate);
+    isAllowedSuccessMetricName = MetricRegistry.name(AlwaysAllowDelegatingPermissionCheck.class, "isAllowed", "success", "histogram");
+    isAllowedFailureMetricName = MetricRegistry.name(AlwaysAllowDelegatingPermissionCheck.class, "isAllowed", "failure", "histogram");
+    checkAllowedOrThrowSuccessMetricName = MetricRegistry.name(AlwaysAllowDelegatingPermissionCheck.class, "checkAllowedOrThrow", "success", "histogram");
+    checkAllowedOrThrowExceptionMetricName = MetricRegistry.name(AlwaysAllowDelegatingPermissionCheck.class, "checkAllowedOrThrow", "exception", "histogram");
```

Comment:
Ah, something super subtle here:

So, we're using the same method of constructing the metric names as the actual source code (invoking MetricRegistry.name()). Now, imagine there's a bug that's introduced into that method which changes how the metric names are built: since the test and the code both use it they won't notice a difference, but the actual emitted metric will be different (which could throw off some alarms). I remember one of the first things I learned at a previous role was that if you're going to do assertions against the final value of a constant expression, you should actually use the literal result as your expected value rather than reconstructing it.

In other words, if MetricRegistry.name("foo", "bar") results in the metric name "foo.bar", you should declare this somewhere in the test:

```
fooBarMetricName = "foo.bar";
```

That way the expected result is hard-coded and immune to changes in how the code works, so you'll know right away if something breaks.

Separately, if these are constants that are used across multiple tests there's no need to keep re-declaring them in @ Before (which is the per-test setup method). You could declare them as plain ol' private static final constants, or you could make them private static and use a static method marked with @ BeforeClass to assign them once for the whole test class. A method (needs to be static, no args, public, returns void) marked with @ BeforeClass is called once before any of the tests in the current test class are run.